ui-tests/src/test/java/org/openmrs/reference/DeleteFormTest.java

```diff
@@ -23,14 +24,14 @@
     @Before
     public void setUp() throws Exception {
 
-        homePage = new HomePage(page);
-        assertPage(homePage);
```

Comment:
Any reason for removing this function

Reply:
This function is very important as it checks the current page. Did you take a look at this [review ](https://github.com/openmrs/openmrs-distro-referenceapplication/pull/421). This `assertPage() ` should not omitted but work around ensuring the test passes with this function present rather than removing it

Reply:
Please check my comment above

Reply:
> Please check my comment above

work around ensuring the test passes with the presence of this function rather than getting rid of it.

Reply:
Add waitForPage() function on homePage object to ensure the page is loaded for assertion to be effective ie assertPage(homePage.waitForPage()); otherwise the function asserts an empty page resulting into java.lang.AssertionError.

Reply:
; otherwise the function asserts an empty page resulting into java.lang.AssertionError., what are you referring here

Reply:
> what are you referring here

By the time the function applies its logic to assert the presence of the page, unfortunately the `homepage` by that time is not yet loaded by the `classLoader `and finds `null page` hence `java.laga.AssertionError`. By adding the `waitForPage()` will tell `assertPage()` function to wait to apply its logic until the `page` is loaded

Reply:
Thank you